src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java

```diff
@@ -128,19 +138,60 @@ public UIPipelineDetailsResponse getPipelineDetails(UUID id, Locale locale) thro
             // Need to make sure that all samples are allowed to be updated.
             canUpdateSamples = cart.values()
                     .stream()
-                    .map(samples -> updateSamplePermission.isAllowed(authentication, samples))
-                    .reduce(true, (a, b) -> a && b);
-            detailsResponse.setCanUpdateSamples(canUpdateSamples);
-        }
+					.map(samples -> updateSamplePermission.isAllowed(authentication, samples))
+					.reduce(true, (a, b) -> a && b);
+			detailsResponse.setCanUpdateSamples(canUpdateSamples);
+		}
 
         /*
         Set the acceptable file types
          */
 		detailsResponse.setAcceptsSingleSequenceFiles(description.acceptsSingleSequenceFiles());
 		detailsResponse.setAcceptsPairedSequenceFiles(description.acceptsPairedSequenceFiles());
 
-        return detailsResponse;
-    }
+		/*
+		Dynamic Sources
+		 */
+		if (description.requiresDynamicSource()) {
```

Comment:
Please add comments to this block of code to clarify what's happening

Reply:
Tried updating in a7628b5

Reply:
Very nicely done